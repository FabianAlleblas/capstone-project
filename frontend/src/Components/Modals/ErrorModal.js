import PropTypes from 'prop-types'
import styled from 'styled-components'
import Button from '../Buttons/Button'

ErrorModal.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string,
  closeErrorModal: PropTypes.func.isRequired,
}

export default function ErrorModal({ children, closeErrorModal, className }) {
  return (
    <Wrapper className={className}>
      <Modal>
        <Text>{children}</Text>
        <ModalButton onClick={closeModal}>Okay</ModalButton>
      </Modal>
    </Wrapper>
  )

  function closeModal() {
    closeErrorModal()
  }
}

const Wrapper = styled.div`
  background-color: rgba(0, 0, 0, 0.4);
  display: grid;
  place-items: center;
  width: 100%;
  z-index: 10;
`

const Modal = styled.div`
  background-color: var(--primary-dark);
  border-radius: 20px;
  box-shadow: 4px 4px 10px var(--shadow);
  display: grid;
  gap: 24px;
  padding: 20px;
  width: 75%;
  place-items: center;
`

const Text = styled.p`
  color: var(--form-font-color);
  font-size: 1.25rem;
`

const ModalButton = styled(Button)`
  width: 50%;
`
