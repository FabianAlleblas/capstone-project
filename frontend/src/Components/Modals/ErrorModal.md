```js
<ErrorModal closeErrorModal={() => {}} children={'Error lol!'} />
```
