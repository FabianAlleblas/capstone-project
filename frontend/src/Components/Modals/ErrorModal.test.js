import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import 'jest-styled-components'
import ErrorModal from './ErrorModal'

const closeErrorModalMock = jest.fn()

describe('ErrorModal', () => {
  it('renders correctly', () => {
    const { container } = render(
      <ErrorModal closeErrorModal={closeErrorModalMock}>LoL Error!</ErrorModal>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls closeErrorModalMock correctly', () => {
    const { getByText } = render(
      <ErrorModal closeErrorModal={closeErrorModalMock}>LoL Error!</ErrorModal>
    )
    user.click(getByText('Okay'))
    expect(closeErrorModalMock).toBeCalledWith({})
  })
})
