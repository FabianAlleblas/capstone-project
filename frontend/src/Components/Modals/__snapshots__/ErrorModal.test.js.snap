// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal renders correctly 1`] = `
.c3 {
  background-color: var(--primary-light);
  border-radius: 22px;
  border: none;
  box-shadow: 0 0 4px var(--shadow);
  color: var(--primary-dark);
  font-family: 'Josefin sans';
  font-size: 1.25rem;
  font-weight: 700;
  padding: 12px 16px 8px;
  width: 100%;
}

.c0 {
  background-color: rgba(0,0,0,0.4);
  display: grid;
  place-items: center;
  width: 100%;
  z-index: 10;
}

.c1 {
  background-color: var(--primary-dark);
  border-radius: 20px;
  box-shadow: 4px 4px 10px var(--shadow);
  display: grid;
  gap: 24px;
  padding: 20px;
  place-items: center;
  width: 75%;
}

.c2 {
  color: var(--form-font-color);
  font-size: 1.25rem;
}

.c4 {
  width: 50%;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      LoL Error!
    </p>
    <button
      class="c3 c4"
    >
      Okay
    </button>
  </div>
</div>
`;
